homeassistant/components/moehlenhoff_alpha2/climate.py

```diff
@@ -0,0 +1,179 @@
+"""Support for Alpha2 room control unit via Alpha2 base."""
+import logging
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    HVAC_MODE_AUTO,
+    PRESET_COMFORT,
+    PRESET_ECO,
+    PRESET_NONE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up platform."""
+    return None
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add Alpha2Climate entities from a config_entry."""
+
+    base_update_handler = hass.data[DOMAIN]["connections"][config_entry.entry_id]
+    await base_update_handler.async_update()
+
+    entities = []
+    for heatarea in base_update_handler.base.heatareas:
+        entities.append(Alpha2Climate(base_update_handler, heatarea))
+    async_add_entities(entities)
+
+
+# https://developers.home-assistant.io/docs/core/entity/climate/
+class Alpha2Climate(ClimateEntity):
+    """Alpha2 ClimateEntity."""
+
+    target_temperature_step = 0.2
+
+    def __init__(self, base_update_handler, data):
+        """Initialize Alpha2 ClimateEntity."""
+        self._base_update_handler = base_update_handler
+        self._data = data
+        self._base_update_handler.add_heatarea_update_callback(
```

Comment:
There is a built-in callback registring approach in HA called dispatcher, not sure which component has a clean implementation of this but check it out!

Reply:
Now using HA dispatcher.